### ISSUE TITLE
[BUG] `tailwindcss/no-custom-classname` does not flag non-existing class names when `DEFAULT` is not set

### ISSUE THREAD

**Issue by kachkaev** [francoismassart/eslint-plugin-tailwindcss#265]
**Describe the bug**

If `tailwind.config.js` contains custom theme which does not include `DEFAULT` values, `tailwindcss/no-custom-classname` fails to detect errors.

**To Reproduce**
1. Create a new app with `tailwindcss` and `eslint-plugin-tailwindcss`
2. Configure `tailwindcss/no-custom-classname`
3. Run `pnpm lint`, observe no errors as expected
4. Tweak `tailwind.config.js` [theme](https://tailwindcss.com/docs/theme#customizing-the-default-theme) by adding custom borders without the `DEFAULT` key:
    ```diff
    theme: {
   +  borderWidth: {
   +      foo: '1px',
   +      bar: '2px',
   +    }
      extend: {
   ```
5. Run `pnpm lint` again
6. ❌ Observe that linting for `app/page.js` does not fail despite that are two instances of `className="... border ..."` there. They come from the standard Next.js boilerplate. Because there is no `DEFAULT` border width in the them, `border` is not a valid class name any more, so `tailwindcss/no-custom-classname` produces a false negative.

**Expected behavior**

When using a standard theme or a custom theme with `DEFAULT` `borderWidth`, `borderRadius`, etc. no error is triggered by the rule. Otherwise, unit-less class names like `border`, `rounded`, etc. are flagged.

**Screenshots**

**Environment (please complete the following information):**
 - OS: macOS
 - Softwares + version used: Node v18.16, pnpm 8.6.10

**Additional context**

When I remove `DEFAULT` from the theme, [Tailwind CSS IntelliSense for VS Code](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss) correctly picks the theme and no longer resolves `border` as a valid class name.

#### before custom config

tooltip shows up:
<img width="734" alt="Screenshot 2023-08-17 at 00 13 55" src="https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/608862/a2da8c2f-886f-48ea-ae3c-3d68e6777522">

#### with custom config (no `DEFAULT`)

tooltip is missing (it still shows for other class names)

<img width="757" alt="Screenshot 2023-08-17 at 00 14 30" src="https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/608862/79fba328-04ad-48cb-9c8f-0e81e17ce326">

Interestingly, [tailwindcss/classnames-order](https://github.com/francoismassart/eslint-plugin-tailwindcss/tree/master/docs/rules/classnames-order.md) rule does the right job in this situation: it moves `border` class name to the beginning of the list, because this class name is unrecognised. This means that the config is read correctly.

I managed to reproduce the same bug for `tailwind.config.js` → `theme` → `borderRadius` (`rounded` class name). Other class names may be affected too.

**eslint config file or live demo**

MWE repo: [uploading]

**Comment by francoismassart** [comment]
Closing this issue, cleaning up before release of v4.
